client/extensions/woocommerce/state/action-list/selectors.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Get the current step in the action list.
+ *
+ * @param {Object} rootState The root Calypso state.
+ * @return {Array|null} The action list, or null if no action list is present.
+ */
+export function getActionList( rootState ) {
+	const woocommerce = rootState.extensions.woocommerce;
+	return get( woocommerce, 'actionList', null );
+}
+
+/**
+ * Get the current step in the action list.
+ *
+ * This selector returns the index of the step that is currently executing in the action list.
+ * (The first step in the list that does not have an endTime)
+ *
+ * @param {Object} actionList The action list to check.
+ * @return {Number|null} The index of the current step, or actionList.length if all steps are complete,
+ */
+export function getCurrentStepIndex( actionList ) {
+	for ( let i = 0; i < actionList.length; i++ ) {
```

Comment:
I'm anticipating that in the future we may want more than one action list at a time available to us.

Also, I'm trying to avoid situations where we get the same objects out of root state multiple times. I feel it's best to get the main object we're working with using a single selector, and then use other supporting selectors to get more information. It's just a way of making more options available to us.

It would still be easy to chain together in a one-liner: `getCurrentStepIndex( getActionList( state ) )`